src/types/tests/schema-test.ts

```diff
@@ -628,6 +628,108 @@ describe('schema', () => {
     assert.deepEqual(intersection.fields, schema3.fields);
     assert.deepEqual(intersection.refinement, schema3.refinement);
   }));
+  it('tests schema union, with inline entities', async () => {
+    const manifest = await Manifest.load('./AB.schema',
+      new Loader(null, {
+        './AB.schema': `
+            schema A
+              foo: inline Foo {a: Text, b: Text}
+
+            schema B
+              foo: inline Foo {a: Text}
+            `
+      }));
+    const aType = manifest.findSchemaByName('A');
+    const bType = manifest.findSchemaByName('B');
+
+    deleteFieldRecursively(aType, 'location', {replaceWithNulls: true});
+    deleteFieldRecursively(bType, 'location', {replaceWithNulls: true});
+
+    // Union the names
+    const ABType = aType.toLiteral();
+    ABType.names = ['A', 'B'];
+
+    assert.deepEqual(Schema.union(aType, bType).toLiteral(), ABType);
+    ABType.names = ['B', 'A'];
```

Comment:
Isn't there a way to normalize and compare schemas even if the order is different?

Reply:
Sure, we can normalize and compare the toStrings or we can use the 'type equality'.

The problem with that is that it doesn't give good debugging info, where deepEqualing the literals gives all the information at once.

The only problem that I see with this is that the names are not stored in a set (or otherwise order independent) because some of our code only handles a single (the first) type name (where there have been other attempts to make the type names equally important).

Reply:
Makes sense. I have very little typescript experience. I will rely on your judgement then. :)

Reply:
I appreciate that Gogul, really. (Though it does make me think that we should try to standardize around some common assertion forms when it comes to checking things like this.)